### ISSUE TITLE
Reusing contextual images for frames in a task

### ISSUE THREAD

**Issue by wsykala** [opencv/cvat#6962]
### Actions before raising this issue

- [X] I searched the existing issues and did not find anything similar.
- [X] I read/searched [the docs](https://github.com/cvat-ai/cvat/tree/master#documentation)

### Is your feature request related to a problem? Please describe.

We have a dataset where each task consists of dozens of frames. For each of the frames there are few contextual images that help our annotators.

Because CVAT is not able to reliably handle hundreds of annotations on the UI, as already raised in #3875, and we sometimes have thousands of objects on a single frame, we had to split our dataset into visual grids. We duplicate each frame N times, ask our annotators to label objects in each cell and when they are done they move move to the next cell, which in this case is a new frame. This significantly reduces the visual lag, but as you might imagine this makes the disk storage go up like crazy.

This is not a deal breaker for now, as the dataset is not that big yet and we are using a shared storage option, but as the data grows, we are going to have lots of unnecessary duplicated images. For a dataset of 20 frames with 2 contextual images each, when we apply a grid of 4x4 we have to duplicate each image 16 times, and this includes all contextual images too.

### Describe the solution you'd like

`manifest.jsonl` should have an option to support duplicates for both the images in the dataset and their related images. Currently, with the following manifest:

```
{"version":"1.0"}
{"type":"images"}
{"name":"image1","extension":".jpg","width":720,"height":405,"meta":{"related_images":[]}}
{"name":"image1","extension":".jpg","width":720,"height":405,"meta":{"related_images":[]}}
```

Only one frame will be visible when task is created.

Likewise, when creating a task with this manifest:

```
{"version":"1.0"}
{"type":"images"}
{"name":"image1","extension":".jpg","width":720,"height":405,"meta":{"related_images":["some_file1.jpg"]}}
{"name":"image2","extension":".jpg","width":720,"height":405,"meta":{"related_images":["some_file1.jpg"]}}
```

An Exception will be raised because the format of contextual images directory structure is not met. While I understand the need for directory structure when the data is uploaded to CVAT, I assumed that when fetching all the data from remote fileshare, the entries in manifest file should be absolute, and they should support duplicates

**Comment by bsekachev** [comment]
Hi @wsykala 

> we sometimes have thousands of objects on a single frame,

Are they different classes/shapes?

**Comment by wsykala** [comment]
Hey @bsekachev 

Currently we can have up to 4 classes. Each class is a polygon shape of various sizes. There is an exception for one class where it can be either a polygon shape or a points shape and that is because objects of this type can sometimes be a single pixel on the image.
Most of the time, two classes take the biggest portion of an image, meaning that you could have for example 1600 objects of class 1, 1800 of class 2 and dozens of objects for class 3 and 4.

**Comment by bsekachev** [comment]
@wsykala 

Did you try to use [filters feature](https://opencv.github.io/cvat/docs/manual/advanced/filter/) to show objects only from one class and annotate class 1, then class 2, etc. 

It could help with UI lugs. Generally the same approach you are using, but without duplicating images.

**Comment by wsykala** [comment]
@bsekachev 
We tried, it works great when we need to annotate the object from classes 3 and 4 (dozens of objects only), but unfortunately it is still lagging quite bad with the classes 1 and 2 that both have thousands of shapes on a single frame.

**Comment by wsykala** [comment]
@bsekachev 

After few hours of looking at the code I managed to make it possible to reuse the contextual images (one image can be reused as many times as needed). Checking for all edge cases was quite hard so it works for our specific scenario - all data is coming from a remote filestore only.
I haven't tested it for any other combination, that's why I am not creating a branch with my changes. Maybe you or anyone else from the team can pick up the insights and deliver this feature, or I can try to pick it up in spare time if it's agreed that this is the way that you want to proceed.

The changes:

1. Change the `r+` to `r` in `create` method: https://github.com/opencv/cvat/blob/v2.7.2/utils/dataset_manifest/core.py#L322. This is probably a mistake that should be fixed. We are not writing anything to the manifest file in that function so `r` is sufficient. If it's left as `r+` and one mounts the shared path volume as read-only, currently an Exception will be raised.
2. Add the files to `related_images` mapping in the for loop that iterates over extracted images: https://github.com/opencv/cvat/blob/v2.7.2/cvat/apps/engine/task.py#L995. This needs to be changed, because `related_images` are filled in earlier in the function based on the directory structure, but as I mentioned previously, I adapted the code to fit our needs.
```
related_images[file_name] = properties.get('meta', {}).get('related_images', [])
```
3. `related_images` will now contain duplicates. There is a constraint on the DB schema to not let this happen due to the  `unique_together` parameter in `RelatedFile` model - https://github.com/opencv/cvat/blob/v2.7.2/cvat/apps/engine/models.py#L550. Removing it and applying DB migration allowed us to bypass that constraint. I am not sure if this impacts anything else in the current functionality of CVAT though.

All of these steps work if we assume three things:
1. Contextual images are uploaded to the `related_images` directory. No additional structure is needed.
2. Before creating the task, we provide our own `manifest.jsonl` and it's uploaded along with the images.
3. We set the `sorting_method` to `predefined`. This is crucial, because currently if we don't add it, the mapping is not going to be executed correctly. I don't understand the reason but it was the case.
